<a href="https://colab.research.google.com/github/hansolothe3rd/MACHINE-LEARNING/blob/main/Project_2_Part_1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Project 2 - Part 1
- Daniel Barella
- 12/1/22

In [14]:
import pandas as pd

In [15]:
df = pd.read_csv('/content/Skyserver_12_30_2019 4_49_58 PM.csv')
df.head()

objid          ra        dec         u         g         r  \
0  1237666301628060000   47.372545   0.820621  18.69254  17.13867  16.55555   
1  1237673706652430000  116.303083  42.455980  18.47633  17.30546  17.24116   
2  1237671126974140000  172.756623  -8.785698  16.47714  15.31072  15.55971   
3  1237665441518260000  201.224207  28.771290  18.63561  16.88346  16.09825   
4  1237665441522840000  212.817222  26.625225  18.88325  17.87948  17.47037   

          i         z   run  rerun  camcol  field            specobjid  \
0  16.34662  16.17639  4849    301       5    771  8168632633242440000   
1  17.32780  17.37114  6573    301       6    220  9333948945297330000   
2  15.72207  15.82471  5973    301       1     13  3221211255238850000   
3  15.70987  15.43491  4649    301       3    121  2254061292459420000   
4  17.17441  17.05235  4649    301       3    191  2390305906828010000   

    class  redshift  plate    mjd  fiberid  
0    STAR  0.000115   7255  56597      832  
1    STAR -0.000093   8290  57364      868  
2    STAR  0.000165   2861  54583       42  
3  GALAXY  0.058155   2002  53471       35  
4  GALAXY  0.072210   2123  53793       74

In [16]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 100000 entries, 0 to 99999
Data columns (total 18 columns):
 #   Column     Non-Null Count   Dtype  
---  ------     --------------   -----  
 0   objid      100000 non-null  int64  
 1   ra         100000 non-null  float64
 2   dec        100000 non-null  float64
 3   u          100000 non-null  float64
 4   g          100000 non-null  float64
 5   r          100000 non-null  float64
 6   i          100000 non-null  float64
 7   z          100000 non-null  float64
 8   run        100000 non-null  int64  
 9   rerun      100000 non-null  int64  
 10  camcol     100000 non-null  int64  
 11  field      100000 non-null  int64  
 12  specobjid  100000 non-null  uint64 
 13  class      100000 non-null  object 
 14  redshift   100000 non-null  float64
 15  plate      100000 non-null  int64  
 16  mjd        100000 non-null  int64  
 17  fiberid    100000 non-null  int64  
dtypes: float64(8), int64(8), object(1), uint64(1)
memory usage: 13.7+

# First choice: Sloan Digital Sky Survey DR16

1. Source of data
- https://www.kaggle.com/datasets/muhakabartay/sloan-digital-sky-survey-dr16
2. Brief description of data
- Sloan Digital Sky Survey current DR16 Server Data release with Galaxies, Stars and Quasars.

3. What is the target?
- The variable 'class' identifies an object to be either a galaxy (GALAXY), star (STAR) or quasar (QSO).

4. What does one row represent? (A person?  A business?  An event? A product?)
- A four-color UVGR intermediate-band photometric system (Thuan-Gunn astronomic magnitude system) is discussed in [1]. The Sloan Digital Sky Survey (SDSS) photometric system, a new five-color (u′ g′ r′ i′ z′) wide-band CCD system is described in [2]
- The variables 'run', 'rerun', 'camcol' and 'field' features which describe a field within an image taken by the SDSS. A field is basically a part of the entire image corresponding to 2048 by 1489 pixels. A field can be identified by: - run number, which identifies the specific scan, - the camera column, or "camcol," a number from 1 to 6, identifying the scanline within the run, and the field number. The field number typically starts at 11 (after an initial rampup time), and can be as large as 800 for particularly long runs. - An additional number, rerun, specifies how the image was processed.
5. Is this a classification or regression problem?
- Classification
6. How many features does the data have?
- 17 variables (double) and 1 additional variable (char) 'class'.
7. How many rows are in the dataset?
-  91966
8. What, if any, challenges do you foresee in cleaning, exploring, or modeling with this dataset?
- I don't see any.



In [17]:
df2 = pd.read_csv('/content/water_potability.csv')
df2.head()

ph    Hardness        Solids  Chloramines     Sulfate  Conductivity  \
0       NaN  204.890455  20791.318981     7.300212  368.516441    564.308654   
1  3.716080  129.422921  18630.057858     6.635246         NaN    592.885359   
2  8.099124  224.236259  19909.541732     9.275884         NaN    418.606213   
3  8.316766  214.373394  22018.417441     8.059332  356.886136    363.266516   
4  9.092223  181.101509  17978.986339     6.546600  310.135738    398.410813   

   Organic_carbon  Trihalomethanes  Turbidity  Potability  
0       10.379783        86.990970   2.963135           0  
1       15.180013        56.329076   4.500656           0  
2       16.868637        66.420093   3.055934           0  
3       18.436524       100.341674   4.628771           0  
4       11.558279        31.997993   4.075075           0

In [18]:
df2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3276 entries, 0 to 3275
Data columns (total 10 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   ph               2785 non-null   float64
 1   Hardness         3276 non-null   float64
 2   Solids           3276 non-null   float64
 3   Chloramines      3276 non-null   float64
 4   Sulfate          2495 non-null   float64
 5   Conductivity     3276 non-null   float64
 6   Organic_carbon   3276 non-null   float64
 7   Trihalomethanes  3114 non-null   float64
 8   Turbidity        3276 non-null   float64
 9   Potability       3276 non-null   int64  
dtypes: float64(9), int64(1)
memory usage: 256.1 KB


# Second choice: Water Potability

1. Source of data
- https://www.kaggle.com/datasets/adityakadiwal/water-potability
2. Brief description of data
-The water_potability.csv file contains water quality metrics for 3276 different water bodies.

3. What is the target?
- 'Potability'
4. What does one row represent? (A person?  A business?  An event? A product?)
- Different bodies of water
5. Is this a classification or regression problem?
- Classification
6. How many features does the data have?
- 10 columns 
7. How many rows are in the dataset?
- 3276 rows
8. What, if any, challenges do you foresee in cleaning, exploring, or modeling with this dataset?
- On quick inspection, it appears to be a lil dirty.